jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/services/asn1/message/Asn1DecodeMAPJSON.java

```diff
@@ -1,97 +1,56 @@
+	private ObjectMapper objectMapper = new ObjectMapper();
 
 
 	public Asn1DecodeMAPJSON(OdeProperties odeProps) {
-		super(new StringPublisher(odeProps));
+		super(new StringPublisher(odeProps), odeProps.getMapStartFlag());
 	}
 
 	@Override
-	protected Object process(String consumedData) {
-
-		OdeData odeData = null;
-		OdeMsgPayload payload = null;
-
+	protected OdeAsn1Data process(String consumedData) {
+		OdeAsn1Data messageToPublish = null;
 		try {
-			logger.info("Processing MAP data");
-			logger.debug("MAP data: {}", consumedData);
-			JSONObject rawJSONObject = new JSONObject(consumedData);
-			Set<?> keys = rawJSONObject.keySet();
-			for (Object key : keys) 
-			{				
-				//Send encoded MAP content to Codec service to decode MAP
-				if (key != null && key.toString().equals(MAPContentType)) {
-					/**process consumed data { "MapMessageContent": [{ "metadata": { "utctimestamp:"2020-11-30T23:45:24.913657Z" }, "payload":"00121E38041180000138044CD8EA0F3D3FC4A600000100000000030AE0C0615C18"}]}
-					 */
-					OdeMapMetadata metadata = null;
-          
-					JSONArray rawMAPJsonContentArray = rawJSONObject.getJSONArray(MAPContentType);
-					for (int i = 0; i < rawMAPJsonContentArray.length(); i++) {
-						JSONObject rawMAPJsonContent = (JSONObject) rawMAPJsonContentArray.get(i);
-						String encodedPayload = rawMAPJsonContent.get("payload").toString();
-						JSONObject rawmetadata = (JSONObject) rawMAPJsonContent.get("metadata");
-						logger.debug("RAW MAP: {}", encodedPayload);
-						// construct payload
-						payload = new OdeAsn1Payload(new OdeHexByteArray(encodedPayload));
-
-						// construct metadata
-						metadata = new OdeMapMetadata(payload);
-						metadata.setOdeReceivedAt(rawmetadata.getString("utctimestamp"));
-						metadata.setOriginIp(rawmetadata.getString("originRsu"));
-						metadata.setRecordType(RecordType.mapTx);
-						metadata.setSecurityResultCode(SecurityResultCode.success);
+			JSONObject rawMapJsonObject = new JSONObject(consumedData);
 
-						if (rawmetadata.getString("source").equals("RSU"))
-							metadata.setMapSource(MapSource.RSU);
-						else
-							metadata.setMapSource(MapSource.V2X);
+			String jsonStringMetadata = rawMapJsonObject.get("metadata").toString();
+			OdeMapMetadata metadata = objectMapper.readValue(jsonStringMetadata, OdeMapMetadata.class);
 
-						Asn1Encoding unsecuredDataEncoding = new Asn1Encoding("unsecuredData", "MessageFrame",
-								EncodingRule.UPER);
-						metadata.addEncoding(unsecuredDataEncoding);
+			Asn1Encoding unsecuredDataEncoding = new Asn1Encoding("unsecuredData", "MessageFrame", EncodingRule.UPER);
+			metadata.addEncoding(unsecuredDataEncoding);
 
-						// construct odeData
-						odeData = new OdeAsn1Data(metadata, payload);
+			String payloadHexString = ((JSONObject)((JSONObject) rawMapJsonObject.get("payload")).get("data")).getString("bytes");
+			payloadHexString = super.stripDot2Header(payloadHexString);
```

Comment:
Can we combine the start flags data with this stripDot2Header() behavior into a separate series of classes that is dedicated to remove the Dot2header? Then set this inside AbstractAsn1DecodeMessageJSON without modifying the Message constructor?

Reply:
We will add this to our backlog to get this to be more generic. We really are just wanting to get this functionality to be available so CDOT and WYDOT can both utilize this through DockerHub.

Reply:
@drewjj  Please address the unit test failure reported by sonarscanner

Reply:
This failure seems to only affect Travis. We haven't been able to resolve Travis from failing. Any ideas? @dan-du-car

Reply:
GM, Drew. Just discussed with Dan and Sai. We documented the issue here https://github.com/usdot-jpo-ode/jpo-ode/issues/546 for now and Sai can plan to address this issue with you/Tony in our next sprint in June. For now, we are good with merging the hotfix changes into master, which Dan has reviewed.